meson.build

```diff
@@ -61,4 +61,19 @@ else
     meson.add_dist_script(py, versioneer, '-o', '_version_meson.py')
 endif
 
+if get_option('disable-optimization')
+  add_project_arguments('-DNPY_DISABLE_OPTIMIZATION', language: ['c', 'cpp'])
+  CPU_CONF_BASELINE = 'none'
+  CPU_CONF_DISPATCH = 'none'
+else
+  CPU_CONF_BASELINE = get_option('cpu-baseline')
+  CPU_CONF_DISPATCH = get_option('cpu-dispatch')
+endif
+CPU_CONF_CPREFIX = 'NPY_'
+CPU_CONF_CONFIG = 'npy_cpu_dispatch_config.h'
+CPU_CONF_MULTI_FEATURES = {
+  'AVX2_FMA3': ['AVX2', 'FMA3']
+}
+subdir('meson_cpu')
```

Comment:
There are two subdir of `meson_cpu` the first one is initializing the CPU Dispatcher and the second one builds the dispatch sources. the builder is located in [numpy/numpy/core/meson_cpu](https://github.com/numpy/numpy/blob/8b514f1edf28b01b800b87d4fff7c0c7b34658ea/numpy/core/meson_cpu) and its possible to re-run by copy the entire dir or provides symlink if it needed by other modules e.g. `numpy/random`

> Also, this should probably not be top-level but become numpy/_cpu_features.

For future needs, its better to have it at top-level in case if we decided to move all external dependencies including any sub-modules into new dir `3rdparty`. For example allow building OpenBlas from the source give us the ability to mange their both static and dynamic CPU features by our dispatching mechanism including disable CPU features at the runtime. 

> so how about _cpu_features? Or simply _cpu?
   
how about rename `/meson_cpu` to `/_init_cpu` and `/numpy/core/meson_cpu` to `numpy/core/_build_cpu`?

Reply:
> For future needs, its better to have it at top-level in case if we decided to move all external dependencies including any sub-modules into new dir `3rdparty`.

That's something that is indeed of interest I'd say. I am thinking about this also for SciPy, where there are more dependencies, like Boost. Note that for this, it must be a directory named `subprojects` at the top-level of the repo, see the first sentences under https://mesonbuild.com/Subprojects.html#using-a-subproject.

So to be clear, you are saying here that `meson_cpu` is kind of a separate standalone package, right? So we later on could put it in its own repo, and treat it as a subproject for NumPy/OpenBLAS/SciPy/etc.?

Reply:
> Note that for this, it must be a directory named subprojects

I'm looking to it

> So to be clear, you are saying here that meson_cpu is kind of a separate standalone package, right? So we later on could put it in its own repo, and treat it as a subproject for NumPy/OpenBLAS/SciPy/etc.?

Exactly that's the ultimate goal, and by moving runtime features detection at [core/src/common/npy_cpu_features.c]( https://github.com/numpy/numpy/blob/main/numpy/core/src/common/npy_cpu_features.c) to the package will make it a complete solution.

Reply:
> the second one builds the dispatch sources. the builder is located in [numpy/numpy/core/meson_cpu](https://github.com/numpy/numpy/blob/8b514f1edf28b01b800b87d4fff7c0c7b34658ea/numpy/core/meson_cpu)

The script has been removed and moved its functionality to new module method `mod_features.multi_targets()`.

> So to be clear, you are saying here that meson_cpu is kind of a separate standalone package, right? So we later on could put it in its own repo, and treat it as a subproject for NumPy/OpenBLAS/SciPy/etc.?

Its seems is going to be hard to treat it as `subproject()` due to count `add_project_arguments` to specify baseline arguments and #definitions across all the sources, the current possible way to reuse it is by explicitly import it.

Reply:
> The script has been removed and moved its functionality to new module method `mod_features.multi_targets()`.

This looks much cleaner now I think, thanks.

> Its seems is going to be hard to treat it as `subproject()` due to count `add_project_arguments` to specify baseline arguments and #definitions across all the sources, the current possible way to reuse it is by explicitly import it.

Yeah, I see what you mean there. The way it is now should work and seems pretty clean, just a `subdir('meson_cpu')` at the top and a variable naming convention + a couple of `add_project_argument` calls. Let's resolve this conversation.